plugins/liquid/refined-types-plugin/src/main/kotlin/arrow/meta/plugins/liquid/phases/analysis/solver/collect/ConstraintCollection.kt

```diff
@@ -572,12 +579,36 @@ internal fun Solver.expressionToFormula(
       ex.operationToken.toString() == "EXCLEQ" &&
       ex.right is KtConstantExpression && ex.right?.text == "null" ->
       ex.left?.let { expressionToFormula(it, bindingContext) as? ObjectFormula }?.let { isNotNull(it) }
+    ex is KtBinaryExpression &&
+      ex.operationToken.toString() == "ANDAND" ->
```

Comment:
We should probably extract big bodies within branches to their own functions but I can take care of those on a clean up after I finish isolating away the kotlin compiler apis